system-tests/ft-suite/src/test/kotlin/com/hotels/styx/providers/HealthCheckProviderSpec.kt

```diff
@@ -117,9 +117,11 @@ class HealthCheckProviderSpec : FeatureSpec() {
                     styxServer().routingObject("aaa-02").get().shouldContain("state:inactive")
                 }
 
-                pollOrigins(styxServer, "origin-01", times = 50).let {
-                    withClue("Only the active origin (origin-01) should be taking traffic. Origins distribution: $it") {
-                        it["origin-01"]?:0.shouldBe(50)
+                eventually(2.seconds, AssertionError::class.java) {
+                    pollOrigins(styxServer, times = 50).let {
+                        withClue("Only the active origin (origin-01) should be taking traffic. Origins distribution: $it") {
+                            it["origin-01"] ?: 0.shouldBe(50)
```

Comment:
Please change this (and other similar usages) to `(it["origin-01"] ?: 0).shouldBe(50)`

Reply:
Good catch! Fixed.